engines/crab/TMX/TMXMap.cpp

```diff
@@ -0,0 +1,409 @@
+//{
+//	int pos = 0;
+//	for (auto n = node->first_node("object"); n != nullptr; n = n->next_sibling("object"), ++pos)
+//	{
+//		Vector2i start;
+//		start.load(n);
+//
+//		rapidxml::xml_node<char> *linenode = n->first_node("polyline");
+//		if (linenode != nullptr)
+//		{
+//			Common::String points, x, y;
+//			loadStr(points, "points", linenode);
+//
+//			path.resize(pos + 1);
+//			bool comma = false;
+//			for (auto i = points.begin(); i != points.end(); ++i)
+//			{
+//				if (*i == ',') comma = true;
+//				else if (*i == ' ')
+//				{
+//					path[pos].push_back(GetPoint(start, x, y));
+//					comma = false;
+//					x.clear();
+//					y.clear();
+//				}
+//				else if (comma) y.push_back(*i);
+//				else x.push_back(*i);
+//			}
+//			path[pos].push_back(GetPoint(start, x, y));
+//		}
+//	}
+//}
+//------------------------------------------------------------------------
+// Purpose: Convert point from string to vector
+//------------------------------------------------------------------------
+// const Vector2i TMXMap::GetPoint(const Vector2i &ref, Common::String &x, Common::String &y)
+//{
+//	Vector2i v;
+//	v.x = ref.x + StringToNumber<int>(x);
+//	v.y = ref.y + StringToNumber<int>(y);
+//	return v;
+//}
+
+//------------------------------------------------------------------------
+// Purpose: Clear all data from the level
+//------------------------------------------------------------------------
+void TMXMap::reset() {
+	g_engine->_imageManager->_tileset.reset();
+	_layer.clear();
+
+	_areaNowalk.clear();
+	_areaExit.clear();
+	_areaTrig.clear();
+	_areaStairs.clear();
+	_areaMusic.clear();
+
+	_prop.clear();
+	_spriteLayer = 0;
+}
+
+//------------------------------------------------------------------------
+// Purpose: Draw functions
+//------------------------------------------------------------------------
+void TMXMap::drawDebug(const Rect &camera) {
+	using namespace pyrodactyl::text;
+
+	for (auto i = _areaTrig.begin(); i != _areaTrig.end(); ++i)
+		i->draw(-camera.x, -camera.y, 0, 0, 254, 254);
+
+	for (auto i = _areaExit.begin(); i != _areaExit.end(); ++i)
+		i->_dim.draw(-camera.x, -camera.y, 0, 254, 254, 254);
+
+	for (auto i = _prop.begin(); i != _prop.end(); ++i)
+		i->_pos.draw(-camera.x, -camera.y, 254, 0, 254, 254);
+
+	for (auto i = _areaNowalk.begin(); i != _areaNowalk.end(); ++i)
+		i->draw(-camera.x, -camera.y, 254, 0, 0, 254);
+
+	for (auto i = _areaMusic.begin(); i != _areaMusic.end(); ++i)
+		i->draw(-camera.x, -camera.y, 254, 254, 0, 254);
+
+	for (auto i = _areaStairs.begin(); i != _areaStairs.end(); ++i) {
+		i->draw(-camera.x, -camera.y, 0, 254, 0, 254);
+		//g_engine->_textManager->draw(i->rect.x - camera.x + 100, i->rect.y - camera.y, NumberToString(i->modifier.x), 0);
+		//g_engine->_textManager->draw(i->rect.x - camera.x + 200, i->rect.y - camera.y, NumberToString(i->modifier.y), 0);
+	}
+
+	// Draw the pathfinding grid (SZ)
+	/*for(int x = 0; x < grid->GetDimensions().x; ++x)
```

Comment:
What about restoring this functionality, but make it triggerable via debug console?